src/test.rs

```diff
@@ -144,13 +161,163 @@ fn validation_failure_is_permanent() {
     // temporarily switch to nonblocking to allow us to construct the stream
     // without validating
     stream.set_nonblocking(true).unwrap();
-    let mut stream = tls_stream::Builder::new()
+    let stream = tls_stream::Builder::new()
         .domain("self-signed.badssl.com")
-        .initialize(creds, stream)
-        .unwrap();
+        .initialize(creds, stream);
+    let stream = match stream {
+        Err(HandshakeError::Interrupted(s)) => s,
+        _ => panic!(),
+    };
     stream.get_ref().set_nonblocking(false).unwrap();
-    assert_eq!(stream.write(b"hi").err().unwrap().raw_os_error().unwrap(),
-               winapi::CERT_E_UNTRUSTEDROOT as i32);
-    assert_eq!(stream.write(b"hi").err().unwrap().raw_os_error().unwrap(),
+    let err = unwrap_handshake(stream.handshake().err().unwrap());
+    assert_eq!(err.raw_os_error().unwrap(),
                winapi::CERT_E_UNTRUSTEDROOT as i32);
 }
+
+fn can_trust_localhost_der() -> bool {
+    if env::var("SCHANNEL_RS_SKIP_SERVER_TESTS").is_ok() {
+        return false
+    }
+    let cert = include_bytes!("../test/schannel-ca.der");
+    let cert = CertContext::new(cert).unwrap();
+    let hash_to_find = cert.signature_hash().unwrap();
+    let mut root = CertStore::system("Root").unwrap();
+    let mut my = CertStore::system("My").unwrap();
+    for cert in root.iter().chain(my.iter()) {
```

Comment:
Sure! I'm not actually aware of one (not that I've done a lot of digging), but do you have one in mind?